### PR TITLE
Refactor BlockLoad test

### PR DESCRIPTION
**Pull request by bernhardmgruber**
I was working on a new BlockLoad algorithm (using bulk copy) and did some cleanup on the BlockLoad test.

### CODE REVIEW

File: cub/test/catch2_test_block_load.cu [hunk 1]
```diff
@@ -77,50 +68,45 @@ __global__ void kernel(std::integral_constant<bool, true>, InputIteratorT input,
...
-  kernel<InputIteratorT, OutputIteratorT, ItemsPerThread, ThreadsInBlock, LoadAlgorithm>
-    <<<1, ThreadsInBlock>>>(std::integral_constant<bool, sufficient_resources>{}, input, output, num_items);
+  using block_load_t = cub::BlockLoad<T, ThreadsInBlock, ItemsPerThread, LoadAlgorithm>;
+  using storage_t    = typename block_load_t::TempStorage;
+  constexpr auto sufficient_resources =
+    cuda::std::bool_constant<sizeof(storage_t) <= cub::detail::max_smem_per_block>{};
 
+  c2h::device_vector<T> d_output(d_input.size());
+  kernel<ItemsPerThread, ThreadsInBlock, LoadAlgorithm>
+    <<<1, ThreadsInBlock>>>(sufficient_resources, input, thrust::raw_pointer_cast(d_output.data()), d_input.size());
```
Comment (elstehle) [inline cub/test/catch2_test_block_load.cu:106]: Leaving this here so you don't have to go through the build logs: MSVC build fails due to implicit conversion from `size_t` to `int`.
```suggestion
    <<<1, ThreadsInBlock>>>(sufficient_resources, input, thrust::raw_pointer_cast(d_output.data()), static_cast<int>(d_input.size()));
```

**Review by elstehle — APPROVED** [review]

File: cub/test/catch2_test_block_load.cu [hunk 2]
```diff
@@ -33,37 +33,28 @@
...
-  {
-    return static_cast<int>(threadIdx.x) * ItemsPerThread + item;
-  }
-};
-
-template <int ItemsPerThread, int ThreadsInBlock>
-struct output_idx<ItemsPerThread, ThreadsInBlock, cub::BlockLoadAlgorithm::BLOCK_LOAD_STRIPED>
-{
-  static __device__ int get(int item)
+  if (LoadAlgorithm == cub::BlockLoadAlgorithm::BLOCK_LOAD_STRIPED)
```
Comment (miscco) [inline cub/test/catch2_test_block_load.cu:38]: `if constexpr` when?

No change requested until we require C++17

**Review by miscco — APPROVED** [review]

File: cub/test/catch2_test_block_load.cu (hunk 2 above)
Comment (bernhardmgruber) [reply cub/test/catch2_test_block_load.cu:38]: I had `_CCCL_IF_CONSTEXPR` initially, but then you have to put braces around both branches and add an unreachable in the dead code for nvcc. Too much LOCs for what this function tries to achieve.

Comment (miscco) [reply cub/test/catch2_test_block_load.cu:38]: sad but true

**Comment by bernhardmgruber** [comment]
pre-commit.ci autofix

**Pull request merged**